### PR TITLE
Added type definition for redux-optimistic-ui

### PR DESCRIPTION
**Pull request by asvetliakov**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

File: redux-optimistic-ui/redux-optimistic-ui-tests.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/// <reference path="../redux/redux.d.ts" />
+/// <reference path="./redux-optimistic-ui.d.ts" />
+
+import { Action } from "redux";
+import { OptimisticState, OptimisticAction, BEGIN, optimistic, ensureState, } from "redux-optimistic-ui";
```
Comment (zhengbli) [inline redux-optimistic-ui/redux-optimistic-ui-tests.ts]: Delete the extra `,` at the end

File: redux-optimistic-ui/redux-optimistic-ui.d.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    /**
+     * Enhanced state by redux-optimistic-ui
+     */
+    export interface OptimisticState<TState> {
+        /**
+         * History list
+         * List is Immutable.List<>
+         */
+        history: any;
```
Comment (zhengbli) [inline redux-optimistic-ui/redux-optimistic-ui.d.ts]: If `history` is immutable, maybe making it `readonly` then

Comment (asvetliakov) [inline redux-optimistic-ui/redux-optimistic-ui.d.ts]: I thought it's too earlier to introduce TS 2.0 features at least when the 2.0 is not even beta stage.

Comment (zhengbli) [inline redux-optimistic-ui/redux-optimistic-ui.d.ts]: Aha you are right. It should be added after 2.0 release.

**Comment by zhengbli** [comment]
:+1: Thanks!

**Pull request merged**